include/clap/ext/thread-check.h

```diff
@@ -8,9 +8,24 @@ static CLAP_CONSTEXPR const char CLAP_EXT_THREAD_CHECK[] = "clap.thread-check";
 extern "C" {
 #endif
 
+/* A note on threads as understood by CLAP:
+ *
+ * In the [main-thread], a CLAP plugin may carry out allocations, acquire a mutex and do IO,
+ * bascially anything a reasonably performing program could do.
+ *
+ * Within an [audio-thread] (of which there may be many in a given host), plugins should strive
+ * to meet realtime requirements. I.e. only carry out sufficiently performant and time-bound
+ * operations. So mutexes, memory (de-)allocations, IO, UI rendering should generally be avoided.
+ *
+ * Depending on the host, [main-thread] and [audio-thread] may or may not map to the same
```

Comment:
I would change it to:

* There can be only one main-thread and audio-thread at a time
* The audio-thread may move from one OS thread to another, including the same OS thread as the main-thread

Reply:
>     * The audio-thread may move from one OS thread to another, including the same OS thread as the main-thread

Thanks, I have added this.

>     * There can be only one main-thread and audio-thread at a time

I don't understand that statement.

* At any given point, there may be one main-thread (unlikely to move between OS threads) and multiple audio-threads (quite likely to move between OS threads from a plugin perspective).
* Also, a simple python script that renders a CLAP echo effect onto a WAV file may use only one thread, for which is_main_thread and is_audio_thread are true simultaneously.

BTW, several concurrent data structure implementations make use of caller-provided per-thread tokens. For that, knowing which thread contexts may move between OS threads is crucial, I hope the newest wording is clear enough in that regard.

Reply:
Yes the host can have a thread pool with thousand of threads, but the host can't call process concurrently on the same plugin instance.
So from the plugin perspective, there can be only one audio thread at a time.

```
void process_plugin()
{
   this->audio_thread_id = std::this_thread::get_id();
   // ...
   this->plug->process(this->plug, ...);
   // ...
   this->audio_thread_id = {};
}
```